test/unit/puppetlabs/services/jruby/jruby_puppet_agents_test.clj

```diff
@@ -98,4 +98,16 @@
             jruby-puppet
             jruby-service
             (is (instance? JRubyPuppet jruby-puppet))))
-        (is (= 4 @num-borrows))))))
\ No newline at end of file
+        (is (= 4 @num-borrows))))))
+
+(deftest next-instance-id-test
+  (let [pool-context (jruby-core/create-pool-context
+                       (jruby-testutils/jruby-puppet-config {:max-active-instances 8})
+                       jruby-testutils/default-profiler
+                       jruby-testutils/default-shutdown-fn)]
+    (testing "next instance id should be based on the pool size"
+      (is (= 10 (jruby-agents/next-instance-id 2 pool-context)))
+      (is (= 100 (jruby-agents/next-instance-id 92 pool-context))))
+    (testing "next instance id should wrap after max int"
+      (let [id (- Integer/MAX_VALUE 1)]
+        (is (= (mod id 8) (jruby-agents/next-instance-id id pool-context)))))))
```

Comment:
This seems like unnecessary complication to me.  I don't think negative IDs in log messages is a UX we should care about ... especially since that is very likely to never actually happen (unless somebody manages to run the server for 292 years without restarting it, while handling 1 request/nanosecond the entire time and having `max-requests = 1`).

Also, I was wondering why the ID grows by `pool-size` instead of just 1.